examples/seismic/source.py

```diff
@@ -49,12 +57,47 @@ def __new__(cls, name, grid, ntime=None, npoint=None, data=None,
         # If provided, copy initial data into the allocated buffer
         if data is not None:
             obj.data[:] = data
+
+        # Set the origin and interval in the time axis
+        if time is None:
+            if dt is None or t0 is None:
+                error('Either time or t0 and dt are required to '
+                      'initialise source/receiver objects')
+            else:
+                obj.t0 = t0
+                obj.dt = dt
+        else:
+            obj.t0 = time[0]
+            obj.dt = time[1]-time[0]
+
         return obj
 
     def __init__(self, *args, **kwargs):
         if not self._cached():
             super(PointSource, self).__init__(*args, **kwargs)
 
+    def resample(self, dt, window=None):
+        t0 = self.t0
+        tn = self.t0+(self.data.shape[0])*self.dt
+
+        # The new number of sample points and roundoff adjusted dt
+        nt = int(round((tn-t0)/dt))
+
+        dt = (tn-t0)/nt
+
+        # Create resampled data.
+        data = np.zeros((nt, self.data.shape[1]))
+        for i in range(self.data.shape[1]):
+            data[:, i] = signal.resample(self.data[:, i], nt, window=window)
+
+        # Return new object
+        return PointSource(self.name, self.grid, t0=self.t0, dt=dt, data=data,
+                           coordinates=self.coordinates.data)
+
+    def time(self):
+        return np.linspace(self.t0, self.t0+self.data.shape[0]*self.dt,
```

Comment:
Replaced code - reduced API to reduce chance of error.

Reply:
No I mean if there is `nt` time step and the time step is `dt`, the last time is `(nt-1)*dt` not `nt*dt`

Reply:
It depends on whether you think we should be using:
np.linspace(..., endpoint=False)
or
np.linspace(..., endpoint=True)
I'm advocating endpoint=False because this results in fewer -1's flying about the place.